docs/tutorial/flow-nft-marketplace.md

```diff
@@ -0,0 +1,1804 @@
+              type="text"
+              placeholder="Max"
+              id="nameInput"
+              onChange={setName}
+            />
+            </div>
+            <div>
+            <label for="breedInput">Breed</label>
+            <select className="u-full-width" id="breedInput" onChange={setBreed}>
+              <option value="Labrador">Labrador</option>
+              <option value="Bulldog">Bulldog</option>
+              <option value="Poodle">Poodle</option>
+            </select>
+            </div>
+            <div>
+            <label for="ageInput">Age</label>
+              <select
+                className="u-full-width"
+                id="ageInput"
+                onChange={setAge}
+              >
+                {
+                  [...Array(10).keys()].map(i => <option value={i}>{i}</option>)
+                }
+              </select>
+            </div>
+          </div>
+        <input className="button-primary" type="submit" value="Mint" />
+      </form>
+    </div>
+  );
+};
+
+const style = {
+  padding: '5rem',
+  background: 'white',
+  maxWidth: 350,
+};
+
+export default Form;
+
+```
+
+The `FileSelector.js` component we imported does not yet exist. So, next to `Form.js`, create another component named `FileSelector.js` to handle the image upload logic.
+
+```jsx
+
+// components/FileSelector.js
+
+import { useState } from 'react';
+
+// We are passing `pet` and `setPet` as props to `FileSelector` so we can
+// set the file we selected to the pet state on the `Form` outer scope
+// and keep this component stateless.
+const FileSelector = ({pet, setPet}) => {
+
+  // Read the FileList from the file input component, then
+  // set the first File object to the pet state.
+  const readFiles = (event) => {
+    const files = event.target.files;
+    if (files.length > 0) {
+      setPet({...pet, file: files[0]});
+    }
+  };
+
+  return (
+    <div className="">
+      <label for="fileInput">Image</label>
+      {/* Add readFiles as the onChange handler. */}
+      <input type="file" onChange={readFiles} />
+    </div>
+  );
+};
+
+export default FileSelector;
+
+```
+
+If you import `Form.js` component into `App.js` and insert it anywhere inside the main `App` container, you should see your form that looks similar to the one you see below:
+
+  ![Screenshot of form component](./images/flow-nft-marketplace/form.png)
+
+
+> **💡 How I write code**    
+> The way I write code is to write very small unit at a time, then
+> test that it works as intended, then if it might be reusable, wrap
+> it in a function. Then I repeat.
+>
+> This makes perfect sense. The more code we write, the harder it is
+> to debug and go back to fix something. It is always better to write > less code in general.
+>
+> Here, I would include `console.log()` in each event handler, then
+> test clicking the form to make sure the logged values are what I
+> expected before moving on.
+
+## Minting logic
```

Comment:
In this section (1235-1317) you have explanation-code-explanation. Is there a good way to combine/consolidate explanations so the explanations aren't separated by the code example?

Reply:
Let me work on this.